src/features/core/cookieConsent/ketch/browser.js

```diff
@@ -0,0 +1,83 @@
+import logger from '../../../../utils/logUtil';
+
+/* eslint-disable class-methods-use-this */
+class Ketch {
+  constructor() {
+    this.userConsentedPurposes = [];
+    this.userDeniedPurposes = [];
+
+    // updateKetchConsent callback function to update current consent purpose state
+    // this will be called from ketch ruddersatck plugin
+    window.updateKetchConsent = (consent) => {
+      if (consent.purposes) {
+        const p = consent.purposes;
+        this.userConsentedPurposes = [];
+        this.userDeniedPurposes = [];
+        Object.entries(p).forEach((e) => {
+          if (e[1]) {
+            this.userConsentedPurposes.push(e[0]);
+          } else {
+            this.userDeniedPurposes.push(e[0]);
+          }
+        });
+      }
+    };
+
+    // getKetchUserConsentedPurposes returns current ketch opted-in purposes
+    window.getKetchUserConsentedPurposes = () => this.userConsentedPurposes.slice();
+
+    // getKetchUserDeniedPurposes returns current ketch opted-out purposes
+    window.getKetchUserDeniedPurposes = () => this.userDeniedPurposes.slice();
+
+    // If ketch tag is loaded before rudderstack and has existing consent then update the consent state
+    if (window.ketchConsent) {
+      Object.entries(window.ketchConsent).forEach((e) => {
```

Comment:
what is the type of window.ketchConsent entries? can we denote what is the e[0] and what is the e[1] for the variables to be self explanatory please?

Reply:
window.ketchConsent would be like `{behavioral_advertising: true, email_marketing: true}`